sale_pricelist_global_rule/models/sale_order.py

```diff
@@ -0,0 +1,84 @@
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    need_recompute_pricelist_global = fields.Boolean()
```

Comment:
```suggestion
    need_recompute_pricelist_global = fields.Boolean(store=False, default=True)
```

Reply:
This doesn't work with `store=False`. Odoo handles it similarly in the sale order when the pricelist is changed, by using a storable field. Please see https://github.com/odoo/odoo/blob/3a28e5b0adbb36bdb1155a6854cdfbe4e7f9b187/addons/sale/models/sale_order.py#L288